### PR TITLE
add issue and pr templates

### PR DESCRIPTION
**Pull request by ryanford-dev**
Initial revision/addition of issue and pr templates

reference  #17304

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+...expected behavior
+
+### Describe what actually *is* happening:
+
+...problem behavior
+
+### If the issue related to the rendering or content of the website, please include a screenshot of the problem area.
+
+...screenshot if needed
```
Comment (I3uckwheat) [inline .github/ISSUE_TEMPLATE.md:17]: Should we make a mention here about how they can attach images by just pasting them into github?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
+
+...your text here
+
+### If this PR is related to an open issue please reference it with the `#` operator and the issue number below:
+
+...issue
+
+### If not, please open a new issue describing the problems this PR solves
 
 Thank you,
```
Comment (I3uckwheat) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: Missing "The Odin Project team", like in the Issue template

**Review by I3uckwheat — CHANGES_REQUESTED** [review]

**Comment by I3uckwheat** [comment]
I really like what we have here, it's comprehensive for sure.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (I3uckwheat) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: ON further review, it seems it's there in the original file, but not showing here.

Comment (ryanford-dev) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: It's not missing, it just wasn't a change

**Review by I3uckwheat — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,4 +1,14 @@
-This is a PR template. If you are adding a solution link to the curriculum, leave this as is. If not, delete it and write the message you wish.
+Thanks for your interest in The Odin Project. In order to get PRs closed in a reasonable amount of time, we request that you include a baseline of information about hthe changes you are proposing. Please answer the following triage quesions:
```
Comment (KevinMulhern) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: I see a lot of open source projects providing check lists in their PR templates. Do you think those would be more user friendly for creating PR's?

Maybe something like this at the start:

### Pre Check List
- [ ] [You have read our contributing guide]()
- [ ] This Pull request resolves an existing open issue - [if not please discuss what you want to do with the maintainers first](https://discord.com/channels/505093832157691914/540903304046182425)
etc

I want to add some tooling to the site that will allow contributors to "preview" thier content with our site markdown parser applied. That could easily be added to a checklist like this.

Comment (KevinMulhern) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: I've also noticed we don't have a contributing guide on this repo 🙈 . We did have one at one point. In any case we need something similar to what we have on the web app: https://github.com/TheOdinProject/theodinproject/wiki/Contributing-Guide

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+Thanks for your interest in The Odin Project. As a courtesy to our maintainers please do a search in our issues to make sure this is not a duplicate of an existing issue. In order to get issues addressed in a reasonable amount of time, we request that you include a baseline of information about the issue you're experiencing and how to reproduce it. Please provide the following:
```
Comment (KevinMulhern) [inline .github/ISSUE_TEMPLATE.md:1]: Should we create multiple issue templates? one for feature requests another for bug reports etc.

**Review by KevinMulhern — COMMENTED** [review]
Looking really good @ryanford-dev 💪

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (ryanford-dev) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: @KevinMulhern See #21812 re: contributing guide

There are some other niceties we can add according to the [official guide](https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community) but we can move those discussions to separate issues or the discussion repo.

File: .github/ISSUE_TEMPLATE.md (hunk 4 above)
Comment (ryanford-dev) [reply .github/ISSUE_TEMPLATE.md]: @KevinMulhern great idea. Please review new feature request template, at your leisure.

**Pull request merged**